### PR TITLE
RUC-352 Add versioning to widget initialization

### PR DESCRIPTION
**Pull request by cokuyan**
Some widgets do not have a configuration setup in the backend, so they
must take the main widget's version. This allows those widgets to also
implement versioning.

Tests: manually tested that version is added to iframe as expected

@CrowdTwist/boba 
@georgehlim @pkhedkarcrowdtwist @kchanct 
@c-wright @dkarten

### CODE REVIEW

File: src/widget.coffee [hunk 1]
```diff
@@ -10,13 +10,14 @@
...
-    constructor: ({@src, id, scroll}) ->
+    constructor: ({@src, id, scroll, version}) ->
       @id = parseInt id, 10
       scroll ?= false
 
       # Enforce trailing slash for browser consistency
       # Don't include trailing slash if a query string is present
-      @src += "/" if @src.charAt(@src.length - 1) isnt "/" and @src.indexOf('?') == -1
+      @src += "/" if @src.charAt(@src.length - 1) isnt "/" and @src.indexOf('?') == -1 && !/#\d+/.test(@src)
+      @src += "##{version}" unless /#\d+/.test(@src)
```
Comment (c-wright) [inline src/widget.coffee]: Can you add a comment here that makes it clear what you're checking for and why

**Comment by cokuyan** [comment]
@c-wright updated

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "crowdtwist-widgets-js-sdk",
-  "version": "0.2.9",
+  "version": "0.3.0",
   "contributors": [{
     "name": "Mark Fayngersh",
     "email": "mark@crowdtwist.com"
```
Comment (dkarten) [inline package.json]: out of scope, but should we update this?

Comment (cokuyan) [reply package.json]: eh, he was the original writer, I'd rather he get the ~~shame~~ credit for this code

**Pull request merged**